appscale/tools/appcontroller_client.py

```diff
@@ -471,6 +472,10 @@ def does_user_exist(self, username, silent=False):
           return False
         else:
           raise Exception(user_exists)
+      except BadSecretException as exception:
+        AppScaleLogger.warn("Exception when checking if a user exists: {0}".
+                            format(exception))
+        break
```

Comment:
Returning `None` here seems unusual to me. Is there a reason for this rather than raising an exception or logging and stopping?